code/modules/multiz/movement.dm

```diff
@@ -36,10 +36,19 @@
 		return eyeobj.zMove(direction)
 
 	// Check if we can actually travel a Z-level.
-	if (!can_ztravel())
+	if (!can_ztravel(direction))
 		to_chat(src, "<span class='warning'>You lack means of travel in that direction.</span>")
 		return FALSE
 
+	if (isliving(src))
```

Comment:
i suppose i could do, although the way the function is setup means i'd need to duplicate the can_ztravel function. I couldn't really  think of a clean way to do it by override, how would you do it?